frontend/packages/core/src/card.tsx

```diff
@@ -126,20 +121,104 @@ const BaseCardContent = styled.div<SpacingProps>`
   ${spacing}
 `;
 
-const StyledCardContent = styled(BaseCardContent)({
+const StyledCardContentContainer = styled.div({
   "> .MuiPaper-root": {
     border: "0",
     borderRadius: "0",
   },
+  overflow: "hidden",
+});
+
+const BaseCardActionArea = styled(CardActionArea)<SpacingProps>`
+  ${spacing}
+`;
+
+const StyledCardActionArea = styled(BaseCardActionArea)({
+  ":hover": {
+    backgroundColor: "#F5F6FD",
+  },
+
+  ":active": {
+    backgroundColor: "#D7DAF6",
+  },
 });
 
+const StyledExpandButton = styled(IconButton)({
+  width: "32px",
+  height: "32px",
+  color: "#3548D4",
+});
+
+interface CardContentCollapsibleProps {
+  /** The text to show in the collapse action area when the card content is collapsed/not collapsed.
+   * By default, this is "See Less" for true and "See More" for false.
+   */
+  dualStateTitle: { true: string; false: string };
+  /** The icon to show in the collapse action area when the card content is collapsed/not collapsed.
+   * By default, this is an ArrowUp icon for true and an ArrowDown icon for false.
+   */
+  dualStateIcon: { true: React.ReactElement; false: React.ReactElement };
+}
+
 interface CardContentProps extends SpacingProps {
   children?: React.ReactNode | React.ReactNode[];
+  /** Make the card content collapse at a set max height. The default is false. */
+  collapsible?: boolean;
+  /** The props for the collapse action if collapsible is enabled */
+  collapseAction?: CardContentCollapsibleProps;
+  /** The max height of the card content. The default is none. */
+  maxHeight?: number;
 }
 
-const CardContent = ({ children, ...props }: CardContentProps) => (
-  <StyledCardContent {...props}>{children}</StyledCardContent>
-);
+const CardContent = ({
+  children,
+  collapsible = false,
+  collapseAction = {
+    dualStateTitle: { true: "See Less", false: "See More" },
+    dualStateIcon: { true: <KeyboardArrowUpIcon />, false: <KeyboardArrowDownIcon /> },
+  },
+  maxHeight,
+  ...props
+}: CardContentProps) => {
+  const ref = React.useRef(null);
+  const [showExpand, setShowExpand] = React.useState(false);
+  const [expanded, setExpanded] = React.useState(true);
+
+  const defaultMaxHeight = maxHeight || "none";
+
+  React.useEffect(() => {
+    if (ref.current.scrollHeight > defaultMaxHeight) {
+      setShowExpand(true);
+      setExpanded(false);
+    }
+  }, [defaultMaxHeight]);
+
+  return (
+    <BaseCardContent {...props} ref={ref}>
+      <StyledCardContentContainer style={{ maxHeight: expanded ? "none" : defaultMaxHeight }}>
```

Comment:
nit: should we be adding these styles to the styled component above?

Reply:
oh yeah true, updated!